subscriber.go

```diff
@@ -270,13 +273,7 @@ func NewRedisSubscriber(address string, handler SubscriptionHandler, poolSize in
 		slots:       make([]*redisSubscriberConn, poolSize),
 	}
 	for slot := 0; slot < poolSize; slot++ {
-		// XXX comment from strib: do we have to block on these
-		// subscription calls?  I think the caller might find it
-		// unexpected.  Plus, something that blocks should probably
-		// take a context that can be cancelled by the caller if
-		// things take too long.  Either way, maybe we should document
-		// the blocking behavior in the method comment?
-		subscriber.reconnectSlot(slot, 0)
+		go subscriber.reconnectSlot(slot, 0)
```

Comment:
Oops, missed this on first review: `slot` needs to be saved in a local variable I think, instead of directly passing the loop variable to the goroutine.  From https://golang.org/doc/effective_go.html#channels:

> The bug is that in a Go for loop, the loop variable is reused for each iteration, so the req variable is shared across all goroutines. That's not what we want.